src/server/build-server.ts

```diff
@@ -42,5 +47,36 @@ export function buildServer(): FastifyInstance {
       });
     }
   );
+
+  app.post("/auth", AuthRequestBodySchema, async (request, reply) => {
+    const { email, password } = request.body;
+    const user = await prisma.user.findUnique({
+      where: { email },
+    });
+
+    if (!user) {
+      throw new CustomError("Email not registered on platform", "EML_02");
+    } else {
+      const isPasswordValid = await compare(password, user.password);
+      if (!isPasswordValid) {
+        throw new CustomError("Wrong password. Try again", "PSW_03");
+      }
+    }
+
+    const token = jwt.sign({ id: user.id }, process.env.TOKEN_KEY, {
+      expiresIn: Number(process.env.TOKEN_TIMEOUT) || 30,
```

Comment:
```suggestion
      expiresIn: Number(process.env.TOKEN_TIMEOUT) ?? 30,
```

O operador `??` é mais próprio pra isso, porque ele verifica apenas `null` e `undefined`